tree/dataframe/src/RCsvDS.cxx

```diff
@@ -159,7 +159,7 @@ std::vector<void *> RCsvDS::GetColumnReadersImpl(std::string_view colName, const
    if ((colType == 'd' && typeid(double) != ti) || (colType == 'l' && typeid(Long64_t) != ti) ||
        (colType == 's' && typeid(std::string) != ti) || (colType == 'b' && typeid(bool) != ti)) {
       std::string err = "The type selected for column \"";
-      err += colName;
+      err += std::string(colName);
```

Comment:
This type of change is concerning as this pattern is a very likely to appear in user code.

Reply:
actually that is counter-intuitive ... does the C++ standard really require the creation of a temporary string to concatenate and std::string and a string_view?  This would seem like a waste ...

Reply:
I guess the rationale is that allocations/copies should be explicit and visible

Reply:
`std::experimental::string_view` did not add `string::op+=(string_view)` :-(

Note that before an implicit trip through `TString` happened even here, for `experimental::string_view`.

We can remove this once no more `experimental::string_view` exists.

Reply:
or we can add ```string::op+=(string_view)``` to ```experimental::string_view```

Reply:
We don't add `experimental::string_view`, it's the standard library's implementation we pick up. I am  not sure whether some implementers might have added the operator; mine was libstdc++ from GCC 7.4. I'm hesitant to mess with this. The cost exists but it's bearable: these are - IIUC - not performance hot spots.

Reply:
The problem is not so much performance as backward incompatibility.  I.e. before this patch
```std_string_obj += string_view_obj;``` works on all platform ... even if inadvertently.  After the patch it no longer works ... on some platforms ....

Reply:
I have a compatibility problem too: perfectly valid C++ code breaks the moment we include `TString.h`, for example:
```
   ------------------------------------------------------------------
  | Welcome to ROOT 6.19/01                        https://root.cern |
  | (c) 1995-2019, The ROOT Team; conception: R. Brun, F. Rademakers |
  | Built for linuxx8664gcc on Jan 15 2020, 03:29:00                 |
  | From heads/master@v6-19-01-2427-g97e62a0                         |
  | Try '.help', '.demo', '.license', '.credits', '.quit'/'.q'       |
   ------------------------------------------------------------------

root [0] #include <TString.h>
root [1] std::string_view a = "a";
root [2] std::string b = "b";
root [3] a == b
ROOT_prompt_3:1:3: error: use of overloaded operator '==' is ambiguous (with operand types 'std::string_view' (aka 'basic_string_view<char>') and 'std::string' (aka 'basic_string<char>'))
a == b
~ ^  ~
/cvmfs/sft.cern.ch/lcg/releases/gcc/9.2.0-afc57/x86_64-centos7/bin/../lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../include/c++/9.2.0/string_view:479:5: note: candidate function [with _CharT = char, _Traits = std::char_traits<char>]
    operator==(basic_string_view<_CharT, _Traits> __x,
    ^
/cvmfs/sft-nightlies.cern.ch/lcg/nightlies/dev3/Wed/ROOT/HEAD/x86_64-centos7-gcc9-opt/include/TEveUtil.h:99:6: note: candidate function
bool operator==(const TString& t, const std::string& s);
     ^
/cvmfs/sft-nightlies.cern.ch/lcg/nightlies/dev3/Wed/ROOT/HEAD/x86_64-centos7-gcc9-opt/include/TString.h:752:15: note: candidate function
inline Bool_t operator==(const TString &s1, const TString &s2)
```
Note that this code used to work with ROOT 6.18/04, so here, IMHO, we are talking about fixing a regression.

Another option would be to make the constructor `explicit` only when using `std::string_view` (my case), but leave it implicit when you have to fall back on `experimental::string_view`.

Reply:
Indeed .. I am working to fix both ...